test/libyul/yulOptimizerTests/expressionSimplifier/remove_redundant_shift_masking.yul

```diff
@@ -0,0 +1,10 @@
+{
+    let a := and(0xff, shl(248, calldataload(0)))
+    let b := and(shl(248, calldataload(0)), 0xff)
+}
+// ----
+// expressionSimplifier
+// {
+//     let a := shl(248, calldataload(0))
```

Comment:
Erm, shouldn't that be `0`?

Reply:
Why would that be 0?

Reply:
I mean we optimize away the `AND(0xff` because the lower 8 bits are shifted out (248 = 256-8)

Reply:
`0x100 & 0xff` should be zero.

Reply:
0xff is a number and thus right-aligned, just in case that was the confusion...

Reply:
ah, i confused the shift direction then